pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -168,18 +202,39 @@ pub mod pallet {
 	pub type DomainHookAddress<T: Config> =
 		StorageMap<_, Blake2_128Concat, Domain, [u8; 20], OptionQuery>;
 
-	/// Stores a batch message, not ready yet to be enqueue.
+	/// Stores a batch message, not ready yet to be enqueued.
 	/// Lifetime handled by `start_batch_message()` and `end_batch_message()`
 	/// extrinsics.
 	#[pallet::storage]
 	pub(crate) type PackedMessage<T: Config> =
 		StorageMap<_, Blake2_128Concat, (T::AccountId, Domain), T::Message>;
 
+	/// Storage for pending inbound messages.
+	#[pallet::storage]
+	#[pallet::getter(fn pending_inbound_entries)]
+	pub type PendingInboundEntries<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::SessionId,
+		Blake2_128Concat,
+		(Proof, T::RouterId),
+		InboundEntry<T>,
+	>;
```

Comment:
I think having the `session_id` as key of the map makes that you can not clean the entry later when the same message hash has arrived after the session_id changed.

Why did you choose this instead of something like the solidity implementation?: 
```rust
Map<Proof, T::RouterId, (T::SessionId, InboundEntry)>
```

Reply:
> Why did you choose this instead of something like the solidity implementation?

Mainly due to the aspect of cleaning the map when new routers are set, but will look into it as I said in a different comment.

Reply:
Thanks Cosmin!

Reply:
Done separately [here](https://github.com/centrifuge/centrifuge-chain/pull/1965).